tardis/io/util.py

```diff
@@ -12,6 +13,34 @@
 import logging
 logger = logging.getLogger(__name__)
 
+float_pattern = r"""    # A numeric string consists of:
+    (?P<sign>[-+])?              # an optional sign, followed by either...
+
+    (
+
+        (?=\d|\.\d)              # ...a number (with at least one digit)
+
+        (?P<int>\d*)             # having a (possibly empty) integer part
+
+        (\.(?P<frac>\d*))?       # followed by an optional fractional part
+
+        (E(?P<exp>[-+]?\d+))?    # followed by an optional exponent, or...
```

Comment:
Capital letter `e` for scientific notation is perfectly valid syntax and used widely. Let's not restrict things for no good reason. If we drop the `re.IGNORECASE` we'd have to make this pattern bigger to catch all the variants of `NaN` (`nan`, `NAN`, ...) and `inf`, or restrict syntax and not allow `Inf` (with a capital `i`) for example, which there is no good reason doing that I can think of.